File: rust/src/models/background_task_finished_event.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::background_task_finished_event2::BackgroundTaskFinishedEvent2;

/// Sent when a background task is finished.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct BackgroundTaskFinishedEvent {
    pub data: BackgroundTaskFinishedEvent2,

    pub r#type: String,
}

impl BackgroundTaskFinishedEvent {
    pub fn new(data: BackgroundTaskFinishedEvent2, r#type: String) -> Self {
        Self { data, r#type }
    }
}

File: javascript/src/models/backgroundTaskFinishedEvent.ts
// this file is @generated
import {
  type BackgroundTaskFinishedEvent2,
  BackgroundTaskFinishedEvent2Serializer,
} from "./backgroundTaskFinishedEvent2";

/** Sent when a background task is finished. */
export interface BackgroundTaskFinishedEvent {
  data: BackgroundTaskFinishedEvent2;
  type: string;
}

export const BackgroundTaskFinishedEventSerializer = {
  _fromJsonObject(object: any): BackgroundTaskFinishedEvent {
    return {
      data: BackgroundTaskFinishedEvent2Serializer._fromJsonObject(object["data"]),
      type: object["type"],
    };
  },

  _toJsonObject(self: BackgroundTaskFinishedEvent): any {
    return {
      data: BackgroundTaskFinishedEvent2Serializer._toJsonObject(self.data),
      type: self.type,
    };
  },
};
